test_github_workflows.py

```
@@ -0,0 +1,793 @@
+        dependencies = {}
+        for job_name, job_config in parsed['jobs'].items():
+            needs = job_config.get('needs', [])
+            if isinstance(needs, str):
+                needs = [needs]
+            dependencies[job_name] = needs
+        
+        assert dependencies['lint'] == []
+        assert dependencies['test'] == ['lint']
+        assert set(dependencies['deploy']) == {'lint', 'test'}
+
+    @pytest.mark.parametrize("cron_expression,is_valid", [
```

Ruff (0.11.9)

Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)